application/src/main/java/org/opentripplanner/standalone/config/routerconfig/VectorTileConfig.java

```diff
@@ -164,15 +166,19 @@ public static Layer mapLayer(NodeAdapter node) {
   /**
    * The lowest configured minZoom value of any layer or the fallback of {@link LayerParameters#MIN_ZOOM}
    */
-  public int minZoom() {
-    return layers.stream().mapToInt(LayerParameters::minZoom).min().orElse(MIN_ZOOM);
+  public int minZoom(Set<String> requestedLayers) {
+    return selectLayers(requestedLayers).mapToInt(LayerParameters::minZoom).min().orElse(MIN_ZOOM);
```

Comment:
If there is a layer that is configured to use the default min zoom, for example, in that case we should check what is smaller, the default or the lowest of the configured min zooms. If I read this code right, it only falls back to the default minimum zoom now if none of the layers have a configured min zoom.

Reply:
Or nvm, I think I read the code wrong, it only falls back to the min zoom if there are no layers?

Reply:
Yes, if there are no layers or if none match what is in `requestedLayers`.

Reply:
That was my original plan, but I tried it and the test ends up quite complicated so I will leave it for a follow up. Otherwise the diff will be 50% test infrastructure changes.